sound/soc/sof/fw-file-profile.c

```diff
@@ -192,7 +192,8 @@ sof_file_profile_for_ipc_type(struct snd_sof_dev *sdev,
 	    sof_platform_uses_generic_loader(sdev))
 		ret = sof_test_firmware_file(dev, out_profile, NULL);
 
-	if (!ret)
+	/* No need to test topology file if get_tplg_files is set */
+	if (!ret && !plat_data->machine->get_tplg_files)
```

Comment:
I'm not entirely sure about this, we must have standalone topology file for backwards compatibility and signaling the number of DMICs for example.
The code in topology.c can rely on the fact that this is in place for example.

Reply:
You need to have a match defined in sound/soc/intel/common/ and that includes a topology file field.
Are you planning to leave that to NULL or set it to 'don_t_care'?

and if you have the old sof-bin release which only have the combined file then you don't test it early and the fallback code can only fallback if the _first_ fragment is missing, otherwise it will hard fail.

Reply:
> You need to have a match defined in sound/soc/intel/common/ and that includes a topology file field. Are you planning to leave that to NULL or set it to 'don_t_care'?

My plan is leave it to NULL

> 
> and if you have the old sof-bin release which only have the combined file then you don't test it early and the fallback code can only fallback if the _first_ fragment is missing, otherwise it will hard fail.

In this case, get_tplg_files() will return 0, and we will load the combined file directly. And it won't fallback as tplg_cnt == 1 in this case.

Reply:
On the second thought, set a generic tplg to sof_tplg_filename in the match table will be much easier and seems make more sense.